File: examples/good.ts
function add(a: number, b: number): number {
  return a + b;
}

function sumValues(values: number[]): number {
  return values.reduce((acc, v) => acc + v, 0);
}

function fibonacci(n: number): number {
  if (n < 0) throw new Error("n must be non-negative");
  if (n === 0) return 0;
  let [a, b] = [0, 1];
  for (let i = 1; i < n; i++) {
    [a, b] = [b, a + b];
  }
  return b;
}

File: examples/good.go
package main

import "fmt"

func add(a, b int) int {
	return a + b
}

func sumValues(values []int) int {
	total := 0
	for _, v := range values {
		total += v
	}
	return total
}

func fibonacci(n int) (int, error) {
	if n < 0 {
		return 0, fmt.Errorf("n must be non-negative")
	}
	if n == 0 {
		return 0, nil
	}
	a, b := 0, 1
	for i := 1; i < n; i++ {
		a, b = b, a+b
	}
	return b, nil
}
